Src/coffeescript/cql-execution/src/elm/quantity.coffee

```diff
@@ -19,24 +21,106 @@ module.exports.Quantity = class Quantity extends Expression
+    if other instanceof Quantity and other.unit == @unit
+      @value > parseFloat other.value
+    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration > otherSmallestDuration
+    else null
 
   before: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value < parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit
+      @value < parseFloat other.value
+    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration < otherSmallestDuration
+    else null
     
   equals: (other) ->
-    if other.unit? && other.value? then @unit == other.unit && @value == parseFloat other.value else null
+    if other instanceof Quantity and @unit == other.unit
+      @value == parseFloat other.value
+    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration == otherSmallestDuration
+    else null
 
-time_units = {'years':'year',  'months': 'month',  'days' :'day', 'minutes': 'minute', 'seconds':'seconds', 'milliseconds' : 'millisecond' }
 
 clean_unit = (units) ->
-  if time_units[units] then time_units[units] else units
+  if ucum_time_units[units] then ucum_to_cql_units[ucum_time_units[units]] else units
+
+# Hash of time units and their UCUM equivalents, both case-sensitive and case-insensitive
+# See http://unitsofmeasure.org/ucum.html#para-31
+# The CQL specification says that dates are based on the Gregorian calendar
+ucum_time_units = {'years': 'a_g', 'year': 'a_g', 'YEARS': 'a_g', 'YEAR': 'a_g', 'a_g': 'a_g'
+  , 'a': 'a_j', 'ANN': 'a_j', 'ann': 'a_j', 'A': 'a_j', 'a_j': 'a_j'
+  , 'months': 'mo_g', 'month':'mo_g', 'mo_g': 'mo_g'
+  , 'mo': 'mo_j', 'MO': 'mo_j', 'mo_j', 'mo_j'
+  , 'weeks': 'wk', 'week': 'wk', 'wk': 'wk', 'WK': 'wk'
+  , 'days': 'd', 'day':'d', 'd': 'd', 'D': 'd'
+  , 'hours': 'h', 'hour': 'h', 'h': 'h', 'H': 'h'
+  , 'minutes': 'min', 'minute': 'min', 'min': 'min', 'MIN': 'min'
+  , 'seconds':'s', 'second':'s', 's': 's', 'S': 's'
+  , 'milliseconds' : 'ms', 'millisecond' : 'ms', 'ms': 'ms', 'MS': 'ms'
+  }
+
+ucum_to_cql_units = {
+    'a_j':  'year'
+  , 'a_g':  'year'
+  , 'mo_j': 'month'
+  , 'mo_g': 'month'
+  , 'wk':   'week'
+  , 'd':    'day'
+  , 'h':    'hour'
+  , 's':    'second'
+  , 'min':  'minute'
+  , 'ms':   'millisecond'
+}
+
+get_ucum_units = (units) ->
+  if ucum_time_units[units]
+    ucum_time_units[units]
+  else units
+
+# The smallest common duration is the millisecond
+smallestDuration = (qty) ->
```

Comment:
I would also just add a brief comment saying the function is converting the given value to milliseconds. I know between the name of the function and the existing comment it should be easy to figure out what is going on, but just to make it easier for future developers, I would explicitly say its returning the given quantity.value in milliseconds.